kythe/go/platform/kzip/kzip.go

```diff
@@ -166,16 +156,60 @@ func NewReader(r io.ReaderAt, size int64) (*Reader, error) {
 	} else if fi := archive.File[0].FileInfo(); !fi.IsDir() {
 		return nil, errors.New("archive root is not a directory")
 	}
-
+	root := archive.File[0].Name
+	pref, err := unitPrefix(root, archive.File)
+	if err != nil {
+		return nil, err
+	}
 	return &Reader{
-		zip:  archive,
-		root: archive.File[0].Name,
+		zip:         archive,
+		root:        root,
+		unitsPrefix: pref,
 	}, nil
 }
 
-func (r *Reader) unitPath(digest string) string   { return path.Join(r.root, "units", digest) }
-func (r *Reader) unitPbPath(digest string) string { return path.Join(r.root, "units.pb", digest) }
-func (r *Reader) filePath(digest string) string   { return path.Join(r.root, "files", digest) }
+func unitPrefix(root string, fs []*zip.File) (string, error) {
+	jsonDir := root + prefixJSON + "/"
```

Comment:
```
const (
  jsonDir = root + prefixJSON + "/"
  protoDir = root + prefixProto + "/"
)
```

Reply:
I actually tried that originally, but compiler rejects it (since root is a parameter)